src/averageAges.js

```diff
@@ -33,8 +39,19 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const woman = people.filter(person => person.sex === 'f'
+  && (!withChildren)
+    ? true
+    : people.some(potentialChild => potentialChild.mother === person.name)
+  );
```

Comment:
```suggestion
  const women = people.filter(person =>
    !withChildren
      ? person.sex === 'f'
      : people.some(child => child.mother === person.name) && person.sex === 'f'
  );
```

same story as with 'men'